src/actions/Scroll.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  Action,
+  ActionPreferences,
+  ActionReturnValue,
+  Graph,
+  TypedSelection,
+  SelectionWithEditor,
+} from "../Types";
+import { displaySelectionDecorations } from "../editDisplayUtils";
+import { groupBy } from "../itertools";
+import { commands, TextEditor, Selection, window } from "vscode";
+import { focusEditor } from "./setSelectionsAndFocusEditor";
+
+class Scroll implements Action {
+  targetPreferences: ActionPreferences[] = [{ insideOutsideType: "inside" }];
+
+  constructor(private graph: Graph, private at: string) {
+    this.run = this.run.bind(this);
+  }
+
+  async run([targets]: [TypedSelection[]]): Promise<ActionReturnValue> {
+    const selectionGroups = groupBy(
+      targets,
+      (t: TypedSelection) => t.selection.editor
+    );
+
+    const selections = <SelectionWithEditor[]>[];
+    const lines = <LineWithEditor[]>[];
+
+    [...selectionGroups].forEach(([editor, targets]) => {
+      const { lineNumber, lineNumberDecorate } = getLineNumber(
+        targets,
+        this.at
+      );
+      lines.push({ lineNumber, editor });
+      selections.push({
+        selection: new Selection(lineNumberDecorate, 0, lineNumberDecorate, 0),
+        editor,
+      });
+    });
+
+    await displaySelectionDecorations(
```

Comment:
Ok so after using this for a little bit, I think the UX would be better if we showed the edit decorations first, then scroll, then remove them. We might not even need the decoration sleep timeout. Should be a fairly easy change right?

Reply:
I agree. We still need a sleep timeout because the scrolling itself only takes a few milliseconds and that's not enough to actually see the decoration at all. I implemented it so that time it takes to actually do the proper scrolling is subtracted from the sleep timeout.